crates/napi/src/bindgen_runtime/js_values/object.rs

```diff
@@ -318,23 +318,28 @@ pub trait JsObjectValue<'env>: JsValue<'env> {
     let env = self.value().env;
     #[cfg(feature = "napi5")]
     {
-      let mut closures = properties_iter
-        .clone()
-        .map(|p| p.data)
-        .filter(|data| !data.is_null())
-        .collect::<Vec<*mut std::ffi::c_void>>();
-      let len = Box::into_raw(Box::new(closures.len()));
-      check_status!(unsafe {
-        sys::napi_add_finalizer(
-          env,
-          self.value().value,
-          closures.as_mut_ptr().cast(),
-          Some(finalize_closures),
-          len.cast(),
-          ptr::null_mut(),
-        )
-      })?;
-      std::mem::forget(closures);
+      if !properties.is_empty() {
+        let mut closures = properties_iter
+          .clone()
+          .map(|p| p.data)
+          .filter(|data| !data.is_null())
+          .collect::<Vec<*mut std::ffi::c_void>>();
+        let len = Box::into_raw(Box::new(closures.len()));
```

Comment:
The problem I noticed is that if you allocate this empty box there is still a leak. My try fix was to avoid this allocation altogether and also not add the finalizers if closures was empty.

Is there a requirement to add finalization regardless if you have closures or not?

Reply:
I have `if !properties.is_empty() { }` here, I believe it can avoid the empty Box allocation and useless `napi_add_finalizer`

Reply:
Yes, but properties is not empty. The `closures` vec is empty, because the properties added are plain `name: value ` properties.

So my question is if you have properties, but not closures for them, if you still need to add the finalizers...

My fix is just 

```
if closures.len() > 0 {
 // alloc box
 // add finalizer
}

Reply:
Oh, my bad, will create another pr to fix it